packages/core/src/fhirpath/atoms.ts

```diff
@@ -352,6 +352,43 @@ export class XorAtom extends InfixOperatorAtom {
   }
 }
 
+/**
+ * 6.5.5. implies
+ * Returns true if left is true and right is true,
+ * true left is false and right true, false or empty
+ * true left is empty
+ */
+export class ImpliesAtom extends InfixOperatorAtom {
+  constructor(left: Atom, right: Atom) {
+    super('implies', left, right);
+  }
+
+  eval(context: AtomContext, input: TypedValue[]): TypedValue[] {
+    const leftResult = this.left.eval(context, input);
+    const rightResult = this.right.eval(context, input);
+    if (!this.isValidCollectionLength(leftResult, rightResult)) {
+      throw new Error(
+        `One or more collections are not of length 1 leftResult: ${leftResult.length} rightResult: ${rightResult.length}`
+      );
+    }
+    const leftValue = leftResult.length === 0 ? null : leftResult[0].value;
+    const rightValue = rightResult.length === 0 ? null : rightResult[0].value;
+    if (rightValue === true || leftValue === false || (leftValue === null && rightValue === true)) {
+      return booleanToTypedValue(true);
+    } else if (leftValue === null || rightValue === null) {
+      return [];
+    }
+    return booleanToTypedValue(false);
+  }
+
+  private isValidCollectionLength(leftResult: TypedValue[], rightResult: TypedValue[]): boolean {
+    if (leftResult.length > 1 || rightResult.length > 1) {
+      return false;
+    }
+    return true;
```

Comment:
This can be rewritten more concisely as `return (leftResult.length <= 1 && rightValue.length <= 1);`